File: VB/Grid_BottomNewItemRow/GridDescendant/GridViewInfoRegistratorDescendant.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Base.Handler
Imports DevExpress.XtraGrid.Views.Base.ViewInfo

Namespace Grid_BottomNewItemRow
	Public Class GridViewInfoRegistratorDescendant
		Inherits GridInfoRegistrator

		Public Overrides ReadOnly Property ViewName() As String
			Get
				Return "GridViewDescendant"
			End Get
		End Property
		Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
			Return New GridViewDescendant(TryCast(grid, GridControlDescendant))
		End Function
		Public Overrides Function CreateViewInfo(ByVal view As BaseView) As BaseViewInfo
			Return New GridViewInfoDescendant(TryCast(view, GridViewDescendant))
		End Function

	End Class
End Namespace

File: CS/Grid_BottomNewItemRow/GridDescendant/GridViewInfoRegistratorDescendant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Base.Handler;
using DevExpress.XtraGrid.Views.Base.ViewInfo;

namespace Grid_BottomNewItemRow
{
    public class GridViewInfoRegistratorDescendant : GridInfoRegistrator
    {
        public override string ViewName { get { return "GridViewDescendant"; } }
        public override BaseView CreateView(GridControl grid) { return new GridViewDescendant(grid as GridControlDescendant); }
        public override BaseViewInfo CreateViewInfo(BaseView view) { return new GridViewInfoDescendant(view as GridViewDescendant); }
       
    }
}
